examples/apps/generic_client/generic_client.py

```diff
@@ -483,5 +433,77 @@ def Main(args=None):
             sys.exit(1)
 
 
+def Main(args=None):
+    options = _parse_command_line(args)
+
+    config = _parse_config_file(options.config)
+    if config is None:
+        logger.error("\n Error in parsing config file: {}\n".format(
+            options.config
+        ))
+        sys.exit(-1)
+
+    # mode should be one of listing or registry (default)
+    mode = options.mode
+
+    # Http JSON RPC listener uri
+    uri = options.uri
+    if uri:
+        config["tcf"]["json_rpc_uri"] = uri
+
+    # Address of smart contract
+    address = options.address
+    if address:
+        if mode == "listing":
+            config["ethereum"]["direct_registry_contract_address"] = \
+                address
+        elif mode == "registry":
+            logger.error(
+                "\n Only Worker registry listing address is supported." +
+                "Worker registry address is unsupported \n")
+            sys.exit(-1)
+
+    # worker id
+    worker_id = options.worker_id
+
+    # setup logging
+    config["Logging"] = {
+        "LogFile": "__screen__",
+        "LogLevel": "INFO"
+    }
+
+    plogger.setup_loggers(config.get("Logging", {}))
+    sys.stdout = plogger.stream_to_logger(
+        logging.getLogger("STDOUT"), logging.DEBUG)
+    sys.stderr = plogger.stream_to_logger(
+        logging.getLogger("STDERR"), logging.WARN)
+
+    logger.info("******* Hyperledger Avalon Generic client *******")
+
+    if mode == "registry" and address:
+        logger.error("\n Worker registry contract address is unsupported \n")
+        sys.exit(-1)
+
+    # Retrieve JSON RPC uri from registry list
+    if not uri and mode == "listing":
+        uri = _retrieve_uri_from_registry_list(config)
+        if uri is None:
+            logger.error("\n Unable to get http JSON RPC uri \n")
+            sys.exit(-1)
+
+    # Prepare worker
+    # JRPC request id. Choose any integer value
+    jrpc_req_id = 31
+    worker_registry = JRPCWorkerRegistryImpl(config)
+    if not worker_id:
+        # Get first worker from worker registry
+        worker_id = _lookup_first_worker(worker_registry, jrpc_req_id)
+        if worker_id is None:
+            logger.error("\n Unable to get worker \n")
+            sys.exit(-1)
+
+    _start_client(config, options, jrpc_req_id, worker_id, worker_registry)
```

Comment:
Any specific reason why prepare worker is done outside start_client ?

Reply:
Moved worker prepare code to start_client function.